pontoon/base/tests/test_models.py

```diff
@@ -48,3 +53,85 @@ def test_latest_activity_approved(self):
     def test_latest_activity_none(self):
         """If empty Translation QuerySet, return None."""
         assert_equal(Translation.objects.none().latest_activity(), None)
+
+
+class EntityTests(TestCase):
+    def setUp(self):
+        self.locale = LocaleFactory.create()
+        self.project = ProjectFactory.create(
+            locales=[self.locale]
+        )
+        self.main_resource = ResourceFactory.create(
+            project=self.project,
+            path='main.lang'
+        )
+        self.other_resource = ResourceFactory.create(
+            project=self.project,
+            path='other.lang'
+        )
+        self.main_entity = EntityFactory.create(
+            resource=self.main_resource,
+            string='Source String'
+        )
+        self.other_entity = EntityFactory.create(
+            resource=self.other_resource,
+            string='Other Source String'
+        )
+        self.main_translation = TranslationFactory.create(
+            entity=self.main_entity,
+            locale=self.locale,
+            string='Translated String'
+        )
+        self.other_translation = TranslationFactory.create(
+            entity=self.other_entity,
+            locale=self.locale,
+            string='Other Translated String'
+        )
+        self.subpage = SubpageFactory.create(
+            project=self.project,
+            name='Subpage',
+            resources=[self.main_resource]
+        )
+
+    def assert_entities_equal(self, entity, path, original, translation):
```

Comment:
You're not really comparing two entities... maybe `assert_serialized_entity`? I dunno, it's minor.